packages/mm-backend/src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  HttpException,
+  HttpStatus,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { MixinGateway } from '../../integrations/mixin.gateway';
+import { createHash } from 'crypto';
+import { ConfigService } from '@nestjs/config';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+  private readonly adminPassword: string;
+  constructor(
+    private readonly mixinGateway: MixinGateway,
+    private configService: ConfigService,
+    private jwtService: JwtService,
+  ) {
+    this.adminPassword = configService.get<string>('ADMIN_PASSWORD');
+  }
+
+  async validateUser(password: string): Promise<string> {
+    if (!this.adminPassword || !password) {
+      throw new UnauthorizedException('Password is required');
```

Comment:
This should probably be HTTP Status code: `400`

Reply:
Fixed: 3a4da56a